src/engine/renderer/tr_shade.cpp

```diff
@@ -3143,26 +3143,26 @@ void Tess_End()
 	}
 
 	// for debugging of sort order issues, stop rendering after a given sort value
-	if ( r_debugSort->integer && r_debugSort->integer < tess.surfaceShader->sort )
+	bool skip = r_debugSort->integer && tess.surfaceShader != nullptr
```

Comment:
this nested indentation is really ugly. I'm almost wondering if it's worth it to split out the cleanup stuff below into its own function :-/

Reply:
In an upcoming commit I move the last part to a function like this:
```
void Tess_Clear()
{
	tess.vboVertexSkinning = false;
	tess.vboVertexAnimation = false;

	// clear shader so we can tell we don't have any unclosed surfaces
	tess.multiDrawPrimitives = 0;
	tess.numIndexes = 0;
	tess.numVertexes = 0;

	// This is important after doing CPU-only tessellation with Tess_MapVBOs( true ).
	// A lot of code relies on a behavior of Tess_Begin: automatically map the
	// default VBO *if tess.verts is null*.
	tess.verts = nullptr;
	tess.indexes = nullptr;
}
```

Is that what you mean?

Reply:
Yes.